packages/ingest/test/test-granule.js

```diff
@@ -339,3 +342,161 @@ test('unversionFilename returns filename without version stamp if present', (t)
 
   t.is(expected, actual);
 });
+
+test('handleDuplicateFile throws DuplicateFile if method is called and duplicateHandling is set to "error"', async (t) => {
+  await t.throwsAsync(handleDuplicateFile({
+    duplicateHandling: 'error',
+    target: {
+      Bucket: 'bar',
+      Key: 'foo',
+    },
+  }), { name: 'DuplicateFile' });
+});
+
+test('handleDuplicateFile returns an empty array if duplicateHandling is set to "skip"', async (t) => {
+  const actual = await handleDuplicateFile({
+    duplicateHandling: 'skip',
+  });
+  const expected = [];
+  t.deepEqual(actual, expected);
+});
+
+test('handleDuplicateFile calls syncFileFunction with expected arguments if duplicateHandling is set to "version" and syncFileFunction is provided', async (t) => {
```

Comment:
I'm not convinced we need it, but there is no test for `duplicateHandling is set to "version" and SyncFileFunction is _not_ provided`?

Reply:
Actually, yes.  This needs a short refactor, as it's not checking the s3Object.moveObject call.    Good catch.

Reply:
d1f67afb9